kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsServiceImpl.java

```diff
@@ -0,0 +1,698 @@
+            LOG.debug("getEcustomsFileService().deleteOldDailyDoneFiles() returned " + deleteOldDailyDone);
+            dataFile = getEcustomsFileService().getDailyBatchDataFile(jobRunDate);
+            vendorCountFile = getEcustomsFileService().getDailyBatchVendorCountFile(jobRunDate);
+            batchName = VendorConstants.ECUSTOMS_BATCH_DAILY;
+            List<VendorDetail> vendorDetails = getDailyVendorDetailList(jobRunDate);
+            processVendorDetails = processVendorDetails(jobRunDate, vendorDetails);
+            getEcustomsFileService().createDailyBatchDoneFile(jobRunDate);
+        } catch (Exception e) {
+            LOG.error("Exception: " + e.getMessage());
+            throw new Exception(e.getMessage());
+        }
+        return processVendorDetails;
+    }
+
+    /**
+     * Processes the list of vendors for output.
+     *
+     * @param jobRunDate
+     *
+     * @param vendorDetails
+     * @return
+     * @throws IOException
+     */
+    private boolean processVendorDetails(Date jobRunDate, List<VendorDetail> vendorDetails) throws IOException {
+        FileOutputStream dataFileStream = new FileOutputStream(dataFile);
+        List<String> outputVendorLineList = new ArrayList<String>();
+//        List<VendorDetailExtension> updatedVDEList = new ArrayList<VendorDetailExtension>();
+//        List<Note> noteList = new ArrayList<Note>();
+
+        for (VendorDetail vendorDetail : vendorDetails) {
+            String vendorLine = generateVendorLine(vendorDetail);
+            if (!outputVendorLineList.contains(vendorLine)) {
+                outputVendorLineList.add(vendorLine);
+                dataFileStream.write(vendorLine.getBytes());
+//                VendorDetailExtension updatedVDE = updateVendorDetailExtension(vendorDetail);
+                updateVendor(vendorDetail);
+//                Note note = createNote(vendorDetail);
+//                updatedVDEList.add(updatedVDE);
+//                noteList.add(note);
+            }
+        }
+//        getBusinessObjectService().save(updatedVDEList);
+//        getBusinessObjectService().save(noteList);
+        dataFileStream.flush();
+        dataFileStream.close();
+        writeVendorCountFile(outputVendorLineList.size());
+        KualiInteger listSize = new KualiInteger(outputVendorLineList.size());
+        saveLastRunDate(jobRunDate, listSize);
+        return true;
+    }
+
+    /**
+     * write the file containing number of vendors processed for ecustoms
+     *
+     * @param vendorDetails
+     */
+    private void writeVendorCountFile(int size) throws IOException {
+        FileOutputStream countFileStream = new FileOutputStream(vendorCountFile);
+        String outputMessage = "Vendors added to ecustoms file = " + size;
+        LOG.info(outputMessage);
+        countFileStream.write(outputMessage.getBytes());
+        countFileStream.flush();
+        countFileStream.close();

```

Comment:
Try/Finally here too. And since you get to define the interface, and we really can't do anything else for a bad file/stream close, let's remove the "throws", add in a WARN/ERROR logging statement, and then a IOUtils.closeQuietly() both catch/finally.